src/utils/squid-cache.ts

```diff
@@ -0,0 +1,113 @@
+import type { FindOptionsRelations } from 'typeorm'
+// import { FindOneOptions, EntityClass } from '@subsquid/typeorm-store';
+import { BatchContext, SubstrateBlock } from '@subsquid/substrate-processor'
+import { Store } from '@subsquid/typeorm-store'
+import { FindManyOptions } from '@subsquid/typeorm-store/src/store'
+
+export interface EntityClass<T = any> {
+    new (): T
+}
+
+type CacheEntityParams = [EntityClass<T>, FindOptionsRelations<T>]
+
+class SquidCache {
+    static instance: SquidCache
+
+    /**
+     * Initialize cache entities Map and relations config for fetching data in
+     * load method. Current relations config will be actual for all fetch actions.
+     * Relations will be saved in cache storage like related entities IDs
+     * (e.g. not "token: Token" but "tokenId: string" ) and related entities will
+     * be added to the list for load in the same level as parent entity. In such case,
+     * if same related entity is changed by some logic, this updated related entity will
+     * be available for all parent entities automatically. During Cache.flush all relations
+     * will be updated as whole cache will be pushed to DB.
+     */
+    init(ctx: BatchContext, entityParams: CacheEntityParams[]): void {}
+
+    /**
+     * Get initialized cache instance
+     */
+    static getInstance(): SquidCache {
+        if (!this.instance) this.instance = new SquidCache()
+        return this.instance
+    }
+
+    /**
+     * Add ids of entities which should be loaded, resolved after Cache.load()
+     * (keeps items as Map structure).
+     * If idOrList === '*', fetch all available entities.
```

Comment:
I think it is better just to make `idOrList` optional and return all if it is not sent

Reply:
Good catch.